### PR TITLE
Add admin sign up users features

### PR DESCRIPTION
**Pull request by kevdaci**
## Commit Message Guidelines
```
Header
Blank Line
Body
Blank Line
Footer

The header should look like:
<type>(<scope>): <subject>

The body should have any necessary detailed info about the commit:
An example, references as to where this idea came from, etc.

The footer should have all the issues tagged:
Fixes #123, Fixes #456

So a commit should look like:
feat(users): Add new Yahoo authentication

Yahoo authentication idea proposed by @codydaig
Example implementation in file.js

Fixes #82
```

* Types: 
  * feat - Features, Enhancements, and overall Improvements
  * fix - Fixes, Bugs, HotFixs, etc...
  * doc - Changes to the Documentation and doesn't actually touch any code.
* Scope:
  * The scope should be where the change took place.
  * Examples: users, core, config, articles
* Subject:
  * The subject line should be clear and concise as to what is being accomplished in the commit.
* General Rules:
  * No Line in the Commit message can be longer than 80 characters.
* Reference: [Angular Conventions](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md)

### CODE REVIEW

File: modules/users/client/controllers/admin/add-user.client.controller.js [hunk 1]
```diff
@@ -27,25 +21,30 @@
...
-      // And redirect to the previous or home page
       $state.go($state.previous.state.name || 'home', $state.previous.params);
     }
 
     function onAddUserError(response) {
       Notification.error({ message: response.data.message, title: '<i class="glyphicon glyphicon-remove"></i> Add User Error!', delay: 6000 });
     }
 
-    
+    function alterRole (role) {
```
Comment (RyanClementsHax) [inline modules/users/client/controllers/admin/add-user.client.controller.js:40]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase 

use this function

File: modules/users/client/services/users.client.service.js [hunk 2]
```diff
@@ -87,6 +87,10 @@
       changeToAccepted: {
         method: 'POST',
         url: '/api/unapproved'
+      },
+      adminSignupUser: {
+        method: 'POST',
+        url: '/api/add'
```
Comment (RyanClementsHax) [inline modules/users/client/services/users.client.service.js]: we should change this url to be /api/users/add or /api/users

File: modules/users/server/policies/admin.server.policy.js [hunk 3]
```diff
@@ -25,7 +25,11 @@ exports.invokeRolesPolicies = function () {
     }, {
       resources: '/api/unapproved',
       permissions: '*'
-    }]
+    }, {
+      resources: '/api/add',
```
Comment (RyanClementsHax) [inline modules/users/server/policies/admin.server.policy.js]: read by comment above about the url

File: modules/users/server/routes/admin.server.routes.js [hunk 4]
```diff
@@ -24,6 +24,10 @@ module.exports = function (app) {
     .get(adminPolicy.isAllowed, admin.read)
     .put(adminPolicy.isAllowed, admin.update)
     .delete(adminPolicy.isAllowed, admin.delete);
+
+  // Admin routes for applying a new user.
+  app.route('/api/add')
```
Comment (RyanClementsHax) [inline modules/users/server/routes/admin.server.routes.js]: read my comment above about the url

**Review by RyanClementsHax — CHANGES_REQUESTED** [review]

File: modules/users/client/controllers/admin/add-user.client.controller.js (hunk 1 above)
Comment (kevdaci) [reply modules/users/client/controllers/admin/add-user.client.controller.js:40]: Changing to lowercase or uppercase is not the only problem. The roles shown on the front end are different from the roles in the schema.

Comment (markd315) [reply modules/users/client/controllers/admin/add-user.client.controller.js:40]: Hey, in an HTML dropdown you can have a different value than the text that is shown, like the following:
<option value="ta">TA</option>

Hope this helps, we really shouldn't need to parse it in a seperate JS function. If you're still having trouble doing it the easy way, take a look at my new reassignFromApproval branch for how I did it.

**Review by markd315 — CHANGES_REQUESTED** [review]
I'm so confused, did you start a new branch and pull request for this functionality? It looks like there's a lot of overlap. If so please close this one.

**Review by markd315 — APPROVED** [review]

**Comment by markd315** [comment]
I think Ryan still wants some routing changes but the users still weren't getting approved for the right roles and I had an easier way.

**Review by RyanClementsHax — APPROVED** [review]

**Pull request merged**